cpp/src/parquet/schema.h

```diff
@@ -268,7 +268,7 @@ class PARQUET_EXPORT GroupNode : public Node {
  public:
   // The field_id here is the default to use if it is not set in the SchemaElement
   static std::unique_ptr<Node> FromParquet(const void* opaque_element,
-                                           NodeVector fields = {}, int field_id = -1);
+                                           NodeVector fields = {});
```

Comment:
While we won't use the optional `field_id` argument anymore in Arrow, consumers of this API (that read Parquet files without going through Arrow) may still want to use it for themselves, no?

Reply:
I don't believe so.  There is no reason they can't create the node and then modify the `field_id` after the fact.  The purpose of `FromParquet` is to take the information in the `opaque_element` and put it into the `Node`.  Why allow the `field_id` to be overridden in this way but not allow it for `name` or `repetition`?